mri/reconstruct/utils.py

```diff
@@ -207,19 +211,11 @@ def generate_operators(data, wavelet_name, samples, mu=1e-06, nb_scales=4,
             fourier_op=fourier_op)
 
     # Define the proximity dual/primal operator
-    prox_op = SparseThreshold(linear_op, mu, thresh_type="soft")
+    prox_op = SparseThreshold(Identity(), mu, thresh_type="soft")
 
     # Define the cost function
-    if gradient_space == "synthesis":
-        cost_op = None
-    else:
-        cost_op = DualGapCost(
-            linear_op=linear_op,
-            initial_cost=1e6,
-            tolerance=1e-4,
-            cost_interval=1,
-            test_range=4,
-            verbose=0,
-            plot_output=None)
-
+    cost_op = GenericCost(
```

Comment:
I guess this is the outcome of the whole dual gap conversation right?

Reply:
Yes, this does not implement the dual gap cost, it is just the Generic cost which is gradient cost + proximity cost. 
I will add a TODO here as we arent sure if this is what we want always. Maybe, we can add a parameter to choose between DualGap cost and this

Reply:
and definitely add this in the PR description, it's another change compared to just adding examples.

Like the possibility to have sparse2d wavelets with the correct adjoint (even if it's a temporary fix).

Reply:
Yes, Ill update description